genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaTagServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.core.jpa.services;
+
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.core.jpa.entities.TagEntity;
+import com.netflix.genie.core.jpa.repositories.JpaTagRepository;
+import com.netflix.genie.core.services.TagService;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.validator.constraints.NotBlank;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * JPA based implementation of the TagService interface.
+ *
+ * @author tgianos
+ * @since 3.3.0
+ */
+@Slf4j
+@Transactional
+public class JpaTagServiceImpl implements TagService {
+
+    private final JpaTagRepository tagRepository;
+
+    /**
+     * Constructor.
+     *
+     * @param tagRepository The repository to use to perform CRUD operations on tags
+     */
+    public JpaTagServiceImpl(final JpaTagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+//    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
Need to change this right?

Reply:
No. We decided to leave it as part of the same transaction to avoid risk of cascading connection requests. The save and flush will push the commit however it will include any changes done up to that point in the method stack (since the transaction was opened by entering the initial aspect wrapped method). I believe I went back through the call stack and ensured that the `Tag` and `File` entities were created before any other changes were made but it's something you could help me double check